query-result.hh

```diff
@@ -59,18 +59,18 @@ public:
     // Reserves minimum_result_size and creates new memory accounter for
     // mutation query. Uses the specified maximum result size and may be
     // stopped before reaching it due to memory pressure on shard.
-    future<result_memory_accounter> new_mutation_read(query::max_result_size max_result_size, short_read short_read_allowed);
+    future<result_memory_accounter> new_mutation_read(query::max_result_size max_result_size, short_read short_read_allowed, const query_id& id);
 
     // Reserves minimum_result_size and creates new memory accounter for
     // data query. Uses the specified maximum result size, result will *not*
     // be stopped due to on shard memory pressure in order to avoid digest
     // mismatches.
-    future<result_memory_accounter> new_data_read(query::max_result_size max_result_size, short_read short_read_allowed);
+    future<result_memory_accounter> new_data_read(query::max_result_size max_result_size, short_read short_read_allowed, const query_id& id = query_id{});
```

Comment:
since we have very few callers of `new_data_read` and `new_digest_read`,
I think it'd be better to drop the default param initialization here and below
and expose any call sites that pass a null query_id.